doc/Programming-guide/Using-Kaa-endpoint-SDKs/C++/SDK-Samsung-Artik-5/index.md

```diff
@@ -33,88 +45,89 @@ Install the third-party components as follows.
 
 1. Install the toolchain.
 
+        mkdir artik_root && cd artik_root
+        export ARTIK_ROOT=$(pwd)
         wget https://launchpad.net/linaro-toolchain-binaries/trunk/2013.10/+download/gcc-linaro-arm-linux-gnueabihf-4.8-2013.10_linux.tar.bz2
         tar xvjf gcc-linaro-arm-linux-gnueabihf-4.8-2013.10_linux.tar.bz2
-        sudo mv gcc-linaro-arm-linux-gnueabihf-4.8-2013.10_linux /opt/gcc-linaro-arm-linux-gnueabihf
+        mv gcc-linaro-arm-linux-gnueabihf-4.8-2013.10_linux $ARTIK_ROOT/gcc-linaro-arm-linux-gnueabihf
 
 1. Install zlib (for Avro version higher than 1.7.5).
 
+        cd $ARTIK_ROOT
         wget http://zlib.net/zlib-1.2.8.tar.gz
         tar -xvzf zlib-1.2.8.tar.gz
-        export INSTALLDIR=/opt/gcc-linaro-arm-linux-gnueabihf
+        export INSTALLDIR=$ARTIK_ROOT/gcc-linaro-arm-linux-gnueabihf
         export PATH=$INSTALLDIR/bin:$PATH
         export TARGETMACH=arm-linux-gnueabi
         export BUILDMACH=i686-pc-linux-gnu
         export CROSS=arm-linux-gnueabihf
         export CC=${CROSS}-gcc
         export LD=${CROSS}-ld
         export AS=${CROSS}-as
+        export AR=${CROSS}-ar
         cd zlib-1.2.8
-        ./configure --prefix=/opt/gcc-linaro-arm-linux-gnueabihf/arm-linux-gnueabihf/libc/usr
+        ./configure --prefix=$INSTALLDIR
         make install
 
 1. Install bzip2 (for Avro version higher than 1.7.5).
 
+        cd $ARTIK_ROOT
         wget http://www.bzip.org/1.0.6/bzip2-1.0.6.tar.gz
         tar -xvzf bzip2-1.0.6.tar.gz
-        export INSTALLDIR=/opt/gcc-linaro-arm-linux-gnueabihf
-        export PATH=$INSTALLDIR/bin:$PATH
-        export TARGETMACH=arm-linux-gnueabi
-        export BUILDMACH=i686-pc-linux-gnu
-        export CROSS=arm-linux-gnueabihf
-        export CC=${CROSS}-gcc
-        export LD=${CROSS}-ld
-        export AS=${CROSS}-as
-        export AR=${CROSS}-ar
         cd bzip2-1.0.6
         sed -e "/^all:/s/ test//" Makefile > Makefile-libbz2_so
         make -f Makefile-libbz2_so CC="${CC}" AR="${AR}"
-        make PREFIX=${INSTALLDIR}/arm-linux-gnueabihf/libc/usr install
+        make PREFIX=$INSTALLDIR install
 
 
 1. Install Boost (1.54 or higher).
 
+        cd $ARTIK_ROOT
         wget -O boost_1_59_0.tar.bz2 http://sourceforge.net/projects/boost/files/boost/1.59.0/boost_1_59_0.tar.bz2/download
         tar xvjf boost_1_59_0.tar.bz2
         cd boost_1_59_0
-        export PATH=$PATH:/opt/gcc-linaro-arm-linux-gnueabihf/bin
+        export PATH=$PATH:$INSTALLDIR/bin
         ./bootstrap.sh
          sed -r 's/(using\s+gcc)(\s+;)/\1 : arm : arm-linux-gnueabihf-c++\2/g' -i project-config.jam
-         ./bjam install toolset=gcc-arm --prefix=${INSTALLDIR}
+         ./bjam install toolset=gcc-arm --prefix=$INSTALLDIR
 
+<!--TODO: Provide link to patch in master, not develop -->
```

Comment:
TODO without a ticket number.

Also, provide a link to the current tree (neither master nor develop). (this goes for all patches)